### ISSUE TITLE
[Resource List] hover background leaks resource list when there is no header.

### ISSUE THREAD

**Issue by Flufd** [Shopify/polaris-react#9177]
## Issue summary

https://polaris.shopify.com/components/lists/resource-list

The first item in a resource list that does not have additional items at the top of the list, leaks it's hover background over the rounded corners. The focus ring also does not match the shape of the first item in this case.

<img width="739" alt="image" src="https://user-images.githubusercontent.com/3304890/236916967-972151d7-c4fc-41a2-a4ee-bb303dbebe43.png">

## Expected behavior
The border of the item should be rounded on the first item if there is no extra items at the top of the list (e.g. filtering, search)

## Steps to reproduce the problem

1. First step
1. Second step

## Reduced test case

The best way to get your bug fixed is to provide a reduced test case. This [CodeSandbox template](https://codesandbox.io/s/q82mlq0m26) is a great starting point.

## Specifications

- Are you using the React components? (Y/N):
- Polaris version number:
- Browser:
- Device:
- Operating System:

Or run `npx envinfo --system --binaries --browsers --npmPackages react,react-dom,@shopify/polaris` to provide specifications on your environment including version numbers, browser, device, and operating system.

Paste the results here:

```bash

```
